autoPyTorch/evaluation/tae.py

```diff
@@ -182,6 +169,33 @@ def __init__(
 
         self.search_space_updates = search_space_updates
 
+        if isinstance(self.resampling_strategy, (HoldoutValTypes, CrossValTypes)):
+            eval_function = autoPyTorch.evaluation.train_evaluator.eval_function
+        elif isinstance(self.resampling_strategy, NoResamplingStrategyTypes):
+            eval_function = autoPyTorch.evaluation.fit_evaluator.eval_function
+        else:
+            raise ValueError("Unknown resampling strategy specified."
+                             "Expected resampling strategy to be in "
+                             "'(HoldoutValTypes, CrossValTypes, NoResamplingStrategyTypes)"
+                             "got {}".format(self.resampling_strategy))
+
+        self.worst_possible_result = cost_for_crash
+
+        eval_function = functools.partial(
+            fit_predict_try_except_decorator,
+            ta=eval_function,
+            cost_for_crash=self.worst_possible_result,
+        )
+
+        super().__init__(
+            ta=ta if ta is not None else eval_function,
```

Comment:
It is hard to track where the `ta` is from.
So it is better to put the reference where it is from.
Basically, this difficulty comes from too deep nests, so it is definitely better to solve this matter as well.